paddle/contrib/inference/CMakeLists.txt

```diff
@@ -17,32 +17,21 @@ if(APPLE)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error=pessimizing-move")
 endif(APPLE)
 
-function(inference_api_test TARGET_NAME TEST_SRC)
+function(inference_api_test TARGET_NAME)
     set(options "")
     set(oneValueArgs "")
     set(multiValueArgs ARGS)
     cmake_parse_arguments(inference_test "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
 
     set(PYTHON_TESTS_DIR ${PADDLE_BINARY_DIR}/python/paddle/fluid/tests)
-    set(arg_list "")
+    cc_test(test_paddle_inference_${TARGET_NAME}
+            SRCS test_paddle_inference_${TARGET_NAME}.cc
+            DEPS paddle_fluid_api paddle_inference_api paddle_inference_api_impl
+            ARGS --dirname=${PYTHON_TESTS_DIR}/book/)
     if(inference_test_ARGS)
-        foreach(arg ${inference_test_ARGS})
-            list(APPEND arg_list "_${arg}")
-        endforeach()
-    else()
-        list(APPEND arg_list "_")
+        set_tests_properties(test_paddle_inference_${TARGET_NAME}
+                 PROPERTIES DEPENDS "${inference_test_ARGS}")
```

Comment:
https://github.com/PaddlePaddle/Paddle/blob/e00c1ee10f77ca5e732eab5c43594f1467657e71/paddle/fluid/inference/tests/book/CMakeLists.txt#L22-L23
python单测的名字是test_xxx，而inference_test_ARGS的名字是api_impl，没对上导致依赖没生效。

所以仿照`paddle/fluid/inference/tests/book/CMakeLists.txt`再来改一下就可以了。

Reply:
@luotao1 I think you may not understand this correctly. In the file, I call the function like this:
inference_api_test(api_impl
                   ARGS test_word2vec test_image_classification)
after parsing,
the "inference_test_ARGs" should be "test_word2vec; test_image_classification" which is correct.
when I use:
set_tests_properties(test_paddle_inference_${TARGET_NAME}
                 PROPERTIES DEPENDS "${inference_test_ARGS}")
it will be interpreted as this:
set_tests_properties(test_paddle_inference_api_impl PROPERTIES DEPENDS "test_word2vec; test_image_classification" )

You can read this reference: https://cmake.org/cmake/help/v3.0/module/CMakeParseArguments.html

Reply:
Got it.